api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java

```diff
@@ -391,14 +391,11 @@ public void testCreateUser_otherTextMustBeNullForNonOtherDiscoverySource() {
   public void testSyncDuccVersionStatus_correctVersions() {
     providedWorkbenchConfig.access.currentDuccVersions = ImmutableList.of(3, 4, 5);
 
-    providedWorkbenchConfig.access.currentDuccVersions.forEach(
-        version -> {
-          DbUser user = userDao.findUserByUsername(USERNAME);
-          user.setDuccAgreement(signDucc(user, version));
-          user = userDao.save(user);
-          userService.syncDuccVersionStatus(user, Agent.asSystem());
-          verify(accessModuleService, never()).updateCompletionTime(any(), any(), any());
-        });
+    DbUser user = userDao.findUserByUsername(USERNAME);
+    user.setDuccAgreement(signDucc(user, 3));
```

Comment:
was this change intentional?

Reply:
yes. The previous test does not work because it tried to insert 3 DUCC for the same user without deleting existing entity. It violates a FK violation. If we still want to test all 3, we would need to add DuccDao then delete that old one. So I feel we can just test one actually from the test name. Not sure if we need to test multiple. But I do not have too much context on this, there might be some easier fix?

Reply:
It's normal for a user to sign a version of the DUCC and then later sign a different one, so that's an important use case to test.

Can you say more about the violation that this causes?  I can work on an update to this test if I understand the problem better.

Reply:
Here is the fix: https://github.com/all-of-us/workbench/pull/8527

If you approve it, I will merge it into **this** branch.